src/array/ops/if_else.rs

```diff
@@ -323,37 +323,20 @@ impl ExtensionArray {
     }
 }
 
-impl DateArray {
-    pub fn if_else(&self, other: &Self, predicate: &BooleanArray) -> DaftResult<Self> {
-        let new_array = self.physical.if_else(&other.physical, predicate)?;
-        Ok(Self::new(self.field.clone(), new_array))
-    }
-}
-
-impl TimestampArray {
-    pub fn if_else(&self, other: &Self, predicate: &BooleanArray) -> DaftResult<Self> {
-        let new_array = self.physical.if_else(&other.physical, predicate)?;
-        Ok(Self::new(self.field.clone(), new_array))
-    }
-}
-
-impl EmbeddingArray {
-    pub fn if_else(&self, other: &Self, predicate: &BooleanArray) -> DaftResult<Self> {
-        let new_array = self.physical.if_else(&other.physical, predicate)?;
-        Ok(Self::new(self.field.clone(), new_array))
-    }
-}
-
-impl ImageArray {
-    pub fn if_else(&self, other: &Self, predicate: &BooleanArray) -> DaftResult<Self> {
-        let new_array = self.physical.if_else(&other.physical, predicate)?;
-        Ok(Self::new(self.field.clone(), new_array))
-    }
+macro_rules! impl_logicalarray_if_else {
```

Comment:
Same thing as filter: thoughts on moving this to logicalarray impl: https://github.com/Eventual-Inc/Daft/blob/main/src/datatypes/logical.rs#L15